### ISSUE TITLE
Implementation of the ability to also use the yeedi API

### ISSUE THREAD

**Issue by mrbungle64** [mrbungle64/ecovacs-deebot.js#175]
### The feature request

Implementation of the ability to also use the yeedi API. They seem very similar.

Since I don't own a yeedi device, I need help with the implementation.

### Current status

* Login works
* "yeedi mop station" (`vthpeg`, `K850_PLUS`) [works](#issuecomment-1152975350)
* "yeedi k650" works
* "yeedi 2 hybrid" works
* "yeedi vac hybrid" works

### Additional information

* https://github.com/nicoduj/homebridge-deebotEcovacs/issues/84
* https://github.com/Ligio/hacc-ozmo/issues/38#issuecomment-1146322614

**Comment by mrbungle64** [comment]
@nfacha

Can you please compare this request with the corresponding ("auth/getAuthCode") request from the mobile app?

```
[EcovacsAPI] Calling main api user/getAuthCode with {"uid":"xxx","accessToken":"xxx"}
[EcoVacsAPI] callUserAuthApi calling https://gl-de-openapi.yeedi.com/v1/global/auth/getAuthCode
[EcoVacsAPI] callUserAuthApi config uid=xxx&accessToken=xxx&authTimespan=1654241898113&bizType=ECOVACS_IOT&deviceId=xxx&openId=global&authAppkey=xxx&authSign=xxx
```

I suspect that something is wrong with one of the parameters

**Comment by mrbungle64** [comment]
@nfacha

Thanks to @apfelnutzer the login now works properly 😃 

I deleted the alpha/0.8.3 branch. Please use the master branch now

**Comment by mrbungle64** [comment]
@nfacha

What model do you own?

**Comment by apfelnutzer** [comment]
Did some testing with my Mob Station today(based on `app2.js`) - intermediate results:

**Working:**
- `vacbot.clean();`
- `vacbot.pause();`
- `vacbot.resume();`
- `vacbot.stop();`
- `vacbot.charge();    `   
- `vacbot.spotArea(...);`

**Not working (properly):**
- `vacbot.edge(); `- not supported by model?
- `vacbot.spot();` - performs spot cleaning (1m * 1m) but does not stop engines afterwards - not really supported by model?

**Comment by mrbungle64** [comment]
@apfelnutzer @nfacha

I just released version 0.8.3-alpha.2 (with yeedi support) and I also published this version on npm 😃 
I have also deleted the feature branch. Please use the master branch from now on.

**Comment by Suvero** [comment]
It's possible to implement this library to Homebridge ? I has too Yeedi (2 hybrid) and I trying to add this vac to my homekit.

**Comment by mrbungle64** [comment]
@Suvero 

> It's possible to implement this library to Homebridge ? I has too Yeedi (2 hybrid) and I trying to add this vac to my homekit.

Please post your feedback in the issue that you also created in the homebridge-deebot repo 😉 
https://github.com/bwp91/homebridge-deebot/wiki/Beta-Version

**Comment by mrbungle64** [comment]
@nfacha 

> I have a Yeedi K650
> ...
> I can confirm it is working :)

Thanks for the feedback 👍🏻 

> Also, I assume your node-red lib (https://github.com/mrbungle64/node-red-contrib-ecovacs-deebot) uses this, so this vacuum should work with node-red too?

Yes, but I have not bumped the version of the library yet.
But the next version of the node will have yeedi login support.

**Comment by nfacha** [comment]
> @nfacha
> 
> Have you already tried running some clean commands with the library?

The clean command does work

**Comment by mrbungle64** [comment]
> > @nfacha
> > Have you already tried running some clean commands with the library?
> 
> The clean command does work

Thanks for the feedback 👍🏻 

Have you also tried the [Node-RED node](https://github.com/mrbungle64/node-red-contrib-ecovacs-deebot) yet?

> cd $HOME/.node-red
> npm install https://github.com/mrbungle64/node-red-contrib-ecovacs-deebot.git
>
> You may have to use different path to your Node-RED user data directory.

**Comment by mrbungle64** [comment]
@nfacha 

> Have you also tried the [Node-RED node](https://github.com/mrbungle64/node-red-contrib-ecovacs-deebot) yet?

No feedback for me? 😉

**Comment by mrbungle64** [comment]
@nfacha 

> Also, I assume your node-red lib (https://github.com/mrbungle64/node-red-contrib-ecovacs-deebot) uses this, so this vacuum should work with node-red too?

Do you actually use Node-RED? I just released version 0.4.2 with initial yeedi support.
I would appreciate some feedback 😉

**Comment by mrbungle64** [comment]
* Login works
* "yeedi mop station" works
* "yeedi k650" works
* "yeedi 2 hybrid" works
* "yeedi vac hybrid" works
